packages/@okta/vuepress-site/docs/concepts/third-party-cookies/index.md

```diff
@@ -0,0 +1,49 @@
+
+Okta session cookies that accompany XHR calls to Okta API endpoints like `/sessions/me` and `/user/me`	 are blocked by Safari because they are sent to a different domain than the one the user is on. Okta session cookies never get through to Okta, and Okta cannot complete the request.
+
+The result is that Okta returns 403 forbidden errors to the user or repeatedly directs them back to the sign-in page.
+
+Specifically, this problem affects [certain methods](https://github.com/okta/okta-auth-js#third-party-cookies) of the Okta Auth JavaScript SDK. The [Okta Sign-In Widget](https://github.com/okta/okta-signin-widget#okta-sign-in-widget) uses the affected Okta Auth JavaScript SDK methods internally. Any customer-developed code that directly makes XHR calls to the Okta [Sessions API](/docs/reference/api/sessions/) is also affected. 
+
+### Token renewal in customer-built SPAs that use the OAuth 2.0 implicit flow
+
+If your code uses the OAuth 2.0 implicit flow to handle token renewal (which typically occurs in the context of a SPA that uses the [implicit flow](/docs/guides/implement-implicit/)), Safari can block Okta session cookies from being sent, so that token renewal is never successfully completed.
+
+The result is that ID tokens and access tokens expire without being renewed, and users are prompted to sign in more frequently, with the frequency determined by the token expiry time.
+
+## Workaround: Custom URL Domain
+
+You can implement a workaround by using the [custom URL domain](/docs/guides/custom-url-domain/) feature.
+
+By making your Okta org effectively part of the same domain as your application server from a browser’s perspective, Custom URL Domain moves Okta session cookies to a first-party context. Calls to Okta become calls within the same site, and Safari third-party cookie blocking is no longer triggered.
+
+For example, if your original Okta org is `companyname.okta.com`, and your app server is `app.companyname.com`, you would use the custom URL domain feature to give your Okta org a new URL like `auth.companyname.com`. This puts your app and your Okta org within the same site. It is okay for them to have different subdomains (in this example, `app` and `auth`); it is the domain that needs to be the same (in this example, `companyname.com`) for them to be considered as being on the same site.
+
+After setting up Custom URL Domain, your need to update your configuration of the Okta Sign-In Widget and the the Okta Auth JavaScript SDK, if you use them, so that they use your new custom domain as the base URL of your org. If your code makes any XHR calls directly to Okta endpoints, you need to update the URLs of those calls.
+
+ - For information on updating the configuration of the Okta Sign-In Widget, see <https://github.com/okta/okta-signin-widget#basic-config-options>
```

Comment:
a few examples

### widget
```
 var signIn = new OktaSignIn({
-  baseUrl: 'https://companyname.okta.com'
+  baseUrl: 'https://auth.companyname.com'
 });
```

### authjs
```
 var authClient = new OktaAuth({
-  issuer: 'https://companyname.okta.com'
+  issuer: 'https://auth.companyname.com'
 });
```

### react sdk

```
import { SecureRoute, Security, LoginCallback } from '@okta/okta-react';

class App extends Component {
   render() {
     return (
       <Router>
-        <Security issuer='https://companyname.okta.com/oauth2/default'
+        <Security issuer='https://auth.companyname.com/oauth2/default'
                   clientId='{clientId}'
                   redirectUri={window.location.origin + '/implicit/callback'} >
           <Route path='/' exact={true} component={Home}/>
          <SecureRoute path='/protected' component={Protected}/>
          <Route path='/implicit/callback' component={LoginCallback} />
        </Security>
      </Router>
    );
  }
}

```

### angular sdk
```
import {
  OKTA_CONFIG,
  OktaAuthModule
 } from '@okta/okta-angular';
 
 const oktaConfig = {
-  issuer: 'https://companyname.okta.com/oauth2/default',
+  issuer: 'https://auth.companyname.com/oauth2/default',
   clientId: '{clientId}',
   redirectUri: 'http://localhost:{port}/implicit/callback',
   pkce: true
};

@NgModule({
  imports: [
    ...
    OktaAuthModule
  ],
  providers: [
    { provide: OKTA_CONFIG, useValue: oktaConfig }
  ],
})
export class MyAppModule { }

```

### vue sdk

```
 import Auth from '@okta/okta-vue'
 
 Vue.use(Auth, {
-  issuer: 'https://companyname.okta.com/oauth2/default',
+  issuer: 'https://auth.companyname.com/oauth2/default',
   clientId: '{clientId}',
   redirectUri: 'http://localhost:{port}/implicit/callback',
   scopes: ['openid', 'profile', 'email'],
```

Reply:
Thank you, Haisheng- I will add the examples.

Reply:
@robertdamphousse-okta, could you help confirm that the example code changes above, for a self-hosted widget and for client-side SDKs, should be recommended to customers who are changing over to an Okta custom URL domain? Would customers normally make these changes in source code or in environment variables? Are any values other than these two--org base URL and OAuth2 issuer URL--that need to change when a customer changes over Custom URL domain?

Reply:
Any examples using `localhost` for redirectUri, etc. should probably use `window.location.origin` instead. That way it will work on `localhost` or on whichever domain your app is hosted on (x.companyname.com)

Reply:
All URLs, localhost, production, etc. will need to be added in the app config in the Okta Admin UI

Reply:
Those sample are modified from our SDK readmes actually. How developers actual use them might be different but otherwise I'm not sure what's the best to add sample code except being consistent with our own SDK readme.

Reply:
these look good to me, provided the eng team agrees with the details!

Reply:
> All URLs, localhost, production, etc. will need to be added in the app config in the Okta Admin UI

@aarongranick-okta, I don't think I understand- This is a scenario where a customer starts using custom URL domain--Isn't it just the URL of the Okta org that changes? Would there be any URLs to change in Admin Console?

Reply:
I've added Haisheng's example changes.